src/sys/inotify.rs

```diff
@@ -0,0 +1,120 @@
+use libc::{
+    c_int,
+    uint32_t
+};
+use std::os::unix::io::RawFd;
+use {Error, Result};
+use NixPath;
+use errno::Errno;
+
+pub mod ffi {
+    use libc::{
+        c_char,
+        c_int,
+        uint32_t
+    };
+
+    extern {
+        pub fn inotify_init() -> c_int;
+        pub fn inotify_init1(flags: c_int) -> c_int;
+        pub fn inotify_add_watch(fd: c_int, path: *const c_char, mask: uint32_t) -> c_int;
+        pub fn inotify_rm_watch(fd: c_int, wd: uint32_t) -> c_int;
+    }
+
+    #[derive(Debug, Clone, Copy)]
+    #[repr(C)]
+    pub struct inotify_event {
+        pub wd: c_int,
+        pub mask: uint32_t,
+        pub cookie: uint32_t,
+        pub len: uint32_t
+    }
+}
+
+bitflags! {
+    pub struct EventFlags: uint32_t {
+        const IN_ACCESS         = 0x00000001;
+        const IN_MODIFY         = 0x00000002;
+        const IN_ATTRIB         = 0x00000004;
+        const IN_CLOSE_WRITE    = 0x00000008;
+        const IN_CLOSE_NOWRITE  = 0x00000010;
+        const IN_OPEN           = 0x00000020;
+        const IN_MOVED_FROM     = 0x00000040;
+        const IN_MOVED_TO       = 0x00000080;
+        const IN_CREATE         = 0x00000100;
+        const IN_DELETE         = 0x00000200;
+        const IN_DELETE_SELF    = 0x00000400;
+        const IN_MOVE_SELF      = 0x00000800;
+
+        const IN_UNMOUNT        = 0x00002000;
+        const IN_Q_OVERFLOW     = 0x00004000;
+        const IN_IGNORED        = 0x00008000;
+
+        const IN_ONLYDIR        = 0x01000000;
+        const IN_DONT_FOLLOW    = 0x02000000;
+        const IN_EXCL_UNLINK    = 0x04000000;
+        const IN_MASK_ADD       = 0x20000000;
+        const IN_ISDIR          = 0x40000000;
+        const IN_ONESHOT        = 0x80000000;
+
+        const IN_CLOSE          = Self::IN_CLOSE_WRITE.bits | Self::IN_CLOSE_NOWRITE.bits;
+        const IN_MOVE           = Self::IN_MOVED_FROM.bits | Self::IN_MOVED_TO.bits;
+        const IN_ALL_EVENTS     =
+            Self::IN_ACCESS.bits | Self::IN_MODIFY.bits | Self::IN_ATTRIB.bits | Self::IN_CLOSE_WRITE.bits |
+            Self::IN_CLOSE_NOWRITE.bits | Self::IN_OPEN.bits | Self::IN_MOVED_FROM.bits | Self::IN_MOVED_TO.bits |
+            Self::IN_DELETE.bits | Self::IN_CREATE.bits | Self::IN_DELETE_SELF.bits | Self::IN_MOVE_SELF.bits;
+    }
+}
+
+bitflags! {
+    pub struct InotifyInitFlags: c_int {
+        const IN_CLOEXEC    = 0o02000000;
+        const IN_NONBLOCK   = 0o00004000;
+    }
+}
+
+#[inline]
+pub fn inotify_init() -> Result<RawFd> {
```

Comment:
It looks like `inotify_init1` superseded `inotify_init`.  If so, then Nix should bother with `inotify_init`.  Just use the newer interface.

Reply:
I decided to include it after seeing that Nix exposes both `epoll_create` and `epoll_create1`. Isn't it the same situation ? If I'm not mistaken `epoll_create` was superseded by `epoll_create1` as well.

Reply:
If `inotify_create1` requires a very recent kernel, then that would be a reason to keep `inotify_create` around.  Otherwise, there's probably not much point.